crates/provider-sdk/src/provider.rs

```diff
@@ -780,28 +778,50 @@ impl ProviderConnection {
         ))
     }
 
-    /// Get the provider key that was assigned to this host @ startup
+    /// Get the provider key that was assigned to this host at startup
     #[must_use]
     pub fn provider_key(&self) -> &str {
-        &self.provider_key
+        &self.provider_id
     }
 
-    /// Stores component with link definition
+    /// Stores link in the [ProviderConnection], either as a source link or target link
+    /// depending on if the provider is the source or target of the link
     pub async fn put_link(&self, ld: InterfaceLinkDefinition) {
-        let mut update = self.links.write().await;
-        update.insert(ld.source_id.to_string(), ld);
+        if ld.source_id == self.provider_id {
+            self.source_links
+                .write()
+                .await
+                .insert(ld.target.to_string(), ld);
+        } else {
```

Comment:
The assumption here is that the provider ID is the target right? but what... if it isn't? Do we want to put that case in here, similar to `delete_link`?